app/js/enhancedDate.js

```diff
@@ -1,5 +1,80 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function iife() {
+    
+    // private functions
+    var publicAPI = {
+        setDate: function setDate(userDate) {
+            // Set working date to user input if date object given
+            if (userDate instanceof Date) {
+                var workingDate = userDate;
+            } else {
+                var workingDate = new Date();
+            }
 
+            return workingDate;
+        },
+
+        getDate: function getDate(asDateObj) {
+            // return date object if requested
+            if (asDateObj === true) {
+                return workingDate;
+            } else {
+            // return milliseconds after 1970 by default
+                
+                return workingDate.getTime();
+            }
+        },
+
+        getDayName: function getDayName() {
+            // return weekday name string
+
+            // define lookup for weekday value to name
+            var weekday = { 0: 'Sunday',
+                            1: 'Monday',
+                            2: 'Tuesday',
+                            3: 'Wednesday',
+                            4: 'Thursday',
+                            5: 'Friday',
+                            6: 'Saturday'};
+
+            return weekday[workingDate.getDay()];
+        },
+
+        getMonthName: function getMonthName() {
+            // return month name string
+
+            // define lookup for month value to name
+            var month = {   0: 'January',
```

Comment:
Same deal with this method - better to move it outside the function, but still inside the module.